src/twisted/web/test/test_wsgi.py

```diff
@@ -1848,6 +1851,115 @@ def cbRendered(ignored):
         return d
 
 
+    @inlineCallbacks
+    def test_pendingRequestOperationsRespectConnectionLoss(self):
```

Comment:
this is a huge test with a lot going on there, but no docstring :(

Please consider writing as many docstring as possible so people reading this could assert whether the testing strategy is valid.

Do we really need threads and the real system clock to write this test?

Why not use just a twisted.internet.task.Clock

--------


I guess that you can write this by triggering a write in a thread and before the reactor is spinned, close the connection on the server side... this should trigger the client side connection closing and then when the reactor is spun for the write, it will find that the request is done.